### PR TITLE
[Enhancement] Support str_to_map function transform for trino parser

### PR DESCRIPTION
**Pull request by 839224346**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
- [x] 3.2
- [x] 3.1
- [x] 3.0
- [ ] 2.5

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java [hunk 1]
```diff
@@ -190,6 +190,18 @@ private static void registerStringFunctionTransformer() {
...
         registerFunctionTransformer("length", 1, "char_length", ImmutableList.of(Expr.class));
+
+        // str_to_map(str, del1, del2) -> str_to_map(split(str, del1), del2)
+        registerFunctionTransformer("str_to_map", 3, new FunctionCallExpr("str_to_map",
+                ImmutableList.of(new FunctionCallExpr("split", ImmutableList.of(
+                        new PlaceholderExpr(1, Expr.class), new PlaceholderExpr(2, Expr.class)
+                )), new PlaceholderExpr(3, Expr.class))));
+
+        // str_to_map(str) -> str_to_map(split(str, default_del1), default_del2)
+        registerFunctionTransformer("str_to_map", 3, new FunctionCallExpr("str_to_map",
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java:201]: May these two register steps using same function name cause conflict?

Comment (839224346) [reply fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java]: Yes, fixed.

**Review by kangkaisen — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by andyziye** [comment]
@Mergify backport branch-3.2 branch-3.1 branch-3.0

**Pull request merged**